RiotSwiftUI/Modules/Room/LocationSharing/Coordinator/LocationSharingCoordinator.swift

```diff
@@ -0,0 +1,188 @@
+                    guard let self = self else { return }
+                    
+                    self.parameters.navigationRouter.dismissModule(animated: true, completion: nil)
+                    self.locationSharingViewModel.dispatch(action: .stopLoading(nil))
+                } failure: { [weak self] error in
+                    guard let self = self else { return }
+                    
+                    MXLog.error("[LocationSharingCoordinator] Failed sharing location with error: \(String(describing: error))")
+                    self.locationSharingViewModel.dispatch(action: .stopLoading(error))
+                }
+            }
+            
+        }
+    }
+    
+    func showActivityControllerForLocation(_ location: CLLocationCoordinate2D) {
+        let vc = UIActivityViewController(activityItems: activityItems(location: location),
+                                          applicationActivities: [ShareToMapsAppActivity(type: .apple, location: location),
+                                                                  ShareToMapsAppActivity(type: .google, location: location)])
+        locationSharingHostingController.present(vc, animated: true)
+    }
+    
+    func activityItems(location: CLLocationCoordinate2D) -> [Any] {
+        var items = [Any]()
+        
+        // Make the share sheet show a pretty location thumbnail
+        if let url = NSURL(string: "https://maps.apple.com?ll=\(location.latitude),\(location.longitude)") {
+            items.append(url)
+        }
+        
+        return items
+    }
+}
+
+extension UIActivity.ActivityType {
+    static let shareToMapsApp = UIActivity.ActivityType("Element.ShareToMapsApp")
+}
+
+class ShareToMapsAppActivity: UIActivity {
+    
+    enum MapsAppType {
+        case apple
+        case google
+    }
+    
+    let type: MapsAppType
+    let location: CLLocationCoordinate2D
+    
+    private override init() {
+        fatalError()
+    }
+    
+    init(type: MapsAppType, location: CLLocationCoordinate2D) {
+        self.type = type
+        self.location = location
+    }
+    
+    override var activityTitle: String? {
+        switch type {
+        case .apple:
+            return VectorL10n.locationSharingOpenAppleMaps
+        case .google:
+            return VectorL10n.locationSharingOpenGoogleMaps
+        }
+    }
+    
+    var activityCategory: UIActivity.Category {
+        return .action
+    }
+    
+    override var activityType: UIActivity.ActivityType {
+        return .shareToMapsApp
+    }
+    
+    override func canPerform(withActivityItems activityItems: [Any]) -> Bool {
+        return true
+    }
+    
+    override func prepare(withActivityItems activityItems: [Any]) {
+        var url: URL?
+        switch type {
+        case .apple:
+            url = URL(string: "https://maps.apple.com?ll=\(location.latitude),\(location.longitude)&q=Pin")
```

Comment:
We can probably use a constant or intermediate method for Apple Maps URL as it is used twice